### PR TITLE
Use proper memory queries to rocksdb

### PR DESCRIPTION
**Pull request by NikVolf**
Profiling what RocksDB returns in production revealed some data:

- `rocksdb.estimate-table-readers-mem` and `rocksdb.cur-size-all-mem-tables` are per-column.

- `rocksdb.block-cache-usage` is per database, even so it must be queried via `db.property_int_value_cf` passing any column (equal and shared for all columns) 🤷‍♀️

### CODE REVIEW

File: kvdb-rocksdb/src/lib.rs [hunk 1]
```diff
@@ -232,8 +232,8 @@ struct DBAndColumns {
 	column_names: Vec<String>,
 }
 
-fn static_property_or_warn(db: &DB, prop: &str) -> usize {
-	match db.property_int_value(prop) {
+fn static_property_or_warn(db: &DB, col: &ColumnFamily, prop: &str) -> usize {
```
Comment (ordian) [inline kvdb-rocksdb/src/lib.rs]: can we make it a method on `DBAndColumns`, something like `fn static_property_or_warn(&self, column: u32, property: &str) -> usize`?

File: kvdb-rocksdb/src/lib.rs [hunk 2]
```diff
@@ -244,10 +244,16 @@ fn static_property_or_warn(db: &DB, prop: &str) -> usize {
...
 	fn size_of(&self, ops: &mut parity_util_mem::MallocSizeOfOps) -> usize {
-		self.column_names.size_of(ops)
-			+ static_property_or_warn(&self.db, "rocksdb.estimate-table-readers-mem")
-			+ static_property_or_warn(&self.db, "rocksdb.cur-size-all-mem-tables")
-			+ static_property_or_warn(&self.db, "rocksdb.block-cache-usage")
+		let mut total = self.column_names.size_of(ops);
+		for v in 0..self.column_names.len() {
+			total = total + static_property_or_warn(&self.db, self.cf(v), "rocksdb.estimate-table-readers-mem");
+			total = total + static_property_or_warn(&self.db, self.cf(v), "rocksdb.cur-size-all-mem-tables");
+			if v == 0 {
```
Comment (ordian) [inline kvdb-rocksdb/src/lib.rs]: we can move it out of the loop?

File: kvdb-rocksdb/src/lib.rs [hunk 3]
```diff
@@ -244,10 +244,16 @@ fn static_property_or_warn(db: &DB, prop: &str) -> usize {
 
 impl MallocSizeOf for DBAndColumns {
 	fn size_of(&self, ops: &mut parity_util_mem::MallocSizeOfOps) -> usize {
-		self.column_names.size_of(ops)
-			+ static_property_or_warn(&self.db, "rocksdb.estimate-table-readers-mem")
-			+ static_property_or_warn(&self.db, "rocksdb.cur-size-all-mem-tables")
-			+ static_property_or_warn(&self.db, "rocksdb.block-cache-usage")
+		let mut total = self.column_names.size_of(ops);
+		for v in 0..self.column_names.len() {
+			total = total + static_property_or_warn(&self.db, self.cf(v), "rocksdb.estimate-table-readers-mem");
```
Comment (ordian) [inline kvdb-rocksdb/src/lib.rs]: `+=`

**Comment by NikVolf** [comment]
@dvdplm I will look into it

Approach in this pr offers rather consistent memory report from my observation, so it worth having it meanwhile

**Comment by dvdplm** [comment]
> Approach in this pr offers rather consistent memory report from my observation, so it worth having it meanwhile

Definitely agree we can do this incrementally, but let's make sure we open issues here or in upstream as needed so we don't lose track of what we'd rather do. Thanks!

**Pull request merged**